subprojects/core/src/main/java/org/gradle/api/internal/DependencyInjectingInstantiator.java

```diff
@@ -101,26 +100,30 @@ public CachedConstructor transform(Class<?> aClass) {
         Object[] resolvedParameters = new Object[parameterTypes.length];
         int pos = 0;
         for (int i = 0; i < resolvedParameters.length; i++) {
+            Object currentParameter = null;
```

Comment:
No need to set this to `null`. It's worth declaring this closer to where we assign it, too, i.e. after the business with assigning `targetType` is done.

Reply:
done